lib/jekyll-responsive-image/resize_handler.rb

```diff
@@ -3,21 +3,28 @@ module ResponsiveImage
     class ResizeHandler
       include ResponsiveImage::Utils
 
-      def resize_image(img, config)
-        img.auto_orient! if config['auto_rotate']
-
+      def resize_image(original_image_path, config)
         resized = []
+        original_image_width, original_image_height = ImageSize.path(original_image_path).size
 
         config['sizes'].each do |size|
-          width = size['width']
-          ratio = width.to_f / img.columns.to_f
-          height = (img.rows.to_f * ratio).round
+          original_image_copy = nil
+
+          if config['auto_rotate']
+            original_image_copy = MiniMagick::Image.open(original_image_path)
+            original_image_copy.auto_orient
+            original_image_width, original_image_height = original_image_copy.dimensions
```

Comment:
When `auto_rotate` in enabled, I end up opening the file using MiniMagick, so that I can run `.auto_orient`. `original_image_width` and `_height` are then overwritten by getting them via MiniMagick (discarding the previously gathered information from `ImageSize`).